client/src/lib/RenderReviewList.js

```diff
@@ -0,0 +1,219 @@
+import DataTable, {createTheme} from 'react-data-table-component';
+import React from 'react';
+import {format} from 'date-fns';
+import Select from 'react-select';
+
+const RenderReviewList = () => {
+    let lastFetch;
+    const queryString = new URLSearchParams(window.location.search);
+    const status = queryString.get('status') ?? 'pending';
+    const [data, setData] = React.useState([]);
+    const [loading, setLoading] = React.useState(false);
+    const [totalRows, setTotalRows] = React.useState(0);
+    const [perPage, setPerPage] = React.useState(10);
+    const [currentPage, setCurrentPage] = React.useState(1);
+    const [channelOptions, setChannelOptions] = React.useState([]);
+    const [selectedChannel, setSelectedChannel] = React.useState(queryString.get('channel') ?? 'all');
+    const [selectPlaceHolder, setSelectPlaceHolder] = React.useState('Code Reviews For All Slack Channels');
+
+    const fetchData = async ({channel, limit, page, status, description}) => {
+        // Check last fetch to avoid data refresh during first page loading.
+        if (lastFetch && lastFetch === `${channel}-${limit}-${page}-${status}`) {
+            return;
+        }
+        lastFetch = `${channel}-${limit}-${page}-${status}`;
+        console.log('fetchData', description, lastFetch)
+        setLoading(true);
+        fetch(`/api/reviews/${channel}/${status}?limit=${limit}&page=${page}`)
+            .then((res) => res.json())
+            .then((result) => {
+                setData(result.dataset);
+                setTotalRows(result.total);
+                setLoading(false);
+            });
+    };
+
+    const handlePageChange = page => {
+        console.log('handlePageChange', page);
+        setCurrentPage(page);
+        fetchData({
+            channel: selectedChannel,
+            limit: perPage,
+            page: page,
+            status,
+            description: 'handlePageChange',
+        });
+    };
+
+    const handlePerRowsChange = async (newPerPage, page) => {
+        console.log('handlePerRowsChange', newPerPage, page);
+        setPerPage(newPerPage);
+        fetchData({
+            channel: selectedChannel,
+            limit: newPerPage,
+            page: page,
+            status,
+            description: 'handlePerRowsChange',
+        });
+    };
+
+    const handleChannelSelectionChange = (data) => {
+        console.log('handleChannelSelectionChange', data);
+        const label = channelOptions.find((item) => item.value === data.value).label;
```

Comment:
It's not possible because the data is coming from react-select component build from channelOptions.  But for good practice, yes, it's better to assume otherwise.  Thanks.